addons/orchestrator/templates/_helpers.tpl

```diff
@@ -71,4 +71,118 @@ orchestrator
 {{- else -}}
 {{- printf "orchestrator-%s" .Values.compDefinitionVersionSuffix -}}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+
+{{/*
+Generate configmap
+*/}}
+{{- define "orchestrator.extend.configs" -}}
+{{- range $path, $_ :=  $.Files.Glob "configs/**" }}
+{{ $path | base }}: |-
+{{- $.Files.Get $path | nindent 2 }}
+{{- end }}
+{{- end }}
+
+
+{{/*
+Generate scripts
+*/}}
+{{- define "orchestrator.extend.scripts" -}}
+{{- range $path, $_ :=  $.Files.Glob "scripts/**" }}
+{{ $path | base }}: |-
+{{- $.Files.Get $path | nindent 2 }}
+{{- end }}
+{{- end }}
+
+
+
+{{- define "orchestrator.cmpd.spec.common" -}}
+provider: kubeblocks
+description: orchestrator is a MySQL high availability and replication management tool
+serviceKind: orchestrator
+serviceVersion: 3.2.6
+updateStrategy: BestEffortParallel
+
+roles:
+  - name: primary
+    serviceable: true
+    writable: true
+    votable: true
+  - name: secondary
+    serviceable: true
+    writable: false
+    votable: true
+
+lifecycleActions:
+  roleProbe:
+    builtinHandler: custom
+    customHandler:
+      exec:
+        command:
+          - /bin/bash
+          - -c
+          - |
+            role=$(curl -s http://127.0.0.1:3000/api/leader-check)
+            if [[ $role == "\"OK\"" ]]; then
+              echo -n "primary"
+            elif [[ $role == "\"Not leader\"" ]]; then
+              echo -n "secondary"
+            else
+              echo -n ""
+            fi
+configs:
+  - name: orchestrator-config
+    templateRef: {{ include "orchestrator.componentDefName" . }}-config
+    namespace: {{ .Release.Namespace }}
+    volumeName: configs
+
+scripts:
+  - name: orc-scripts
+    templateRef: {{ include "orchestrator.componentDefName" . }}-scripts
+    namespace: {{ .Release.Namespace }}
+    volumeName: scripts
+    defaultMode: 0555
+
+services:
+  - name: default
+    roleSelector: primary
+    spec:
+      ports:
+        - name: http
+          port: 80
+          targetPort: http
+        - name: prometheus
+          port: 9125
+          targetPort: prometheus
+{{- end }}
+
+
+{{- define "orchestrator.cmpd.spec.runtime.common" -}}
+command:
+  - bash
+  - -c
+  - |
+    /scripts/startup.sh
+volumeMounts:
+  - name: configs
+    mountPath: /configs
+  - name: scripts
+    mountPath: /scripts
+  - mountPath: /data
+    name: data
```

Comment:
Should define the data volume explicitly.

Reply:
used info from .Values.